db/version_set.cc

```diff
@@ -1907,113 +1907,52 @@ Status Version::GetBlob(const ReadOptions& read_options, const Slice& user_key,
 
 void Version::MultiGetBlob(
     const ReadOptions& read_options, MultiGetRange& range,
-    std::unordered_map<uint64_t, BlobReadRequests>& blob_rqs) {
-  if (read_options.read_tier == kBlockCacheTier) {
-    Status s = Status::Incomplete("Cannot read blob(s): no disk I/O allowed");
-    for (const auto& elem : blob_rqs) {
-      for (const auto& blob_rq : elem.second) {
-        const KeyContext& key_context = blob_rq.second;
-        assert(key_context.s);
-        assert(key_context.s->ok());
-        *(key_context.s) = s;
-        assert(key_context.get_context);
-        auto& get_context = *(key_context.get_context);
-        get_context.MarkKeyMayExist();
-      }
-    }
-    return;
-  }
+    std::unordered_map<uint64_t, BlobReadContexts>& blob_ctxs) {
+  assert(!blob_ctxs.empty());
 
-  assert(!blob_rqs.empty());
-  Status status;
+  autovector<BlobFileReadRequests> blob_reqs;
 
-  for (auto& elem : blob_rqs) {
-    const uint64_t blob_file_number = elem.first;
+  for (auto& ctx : blob_ctxs) {
+    const auto file_number = ctx.first;
+    const auto blob_file_meta = storage_info_.GetBlobFileMetaData(file_number);
```

Comment:
We woult still want still error out and adjust the `Status` if the blob file number is invalid (`blob_file_meta` is `nullptr`)

Reply:
I did that inside the loop


```c++
      if (!blob_file_meta) {
        *key_context.s = Status::Corruption("Invalid blob file number");
        continue;
      }
```

Reply:
Right, I see. We should probably rearrange this though because we're potentially deref'ing a null pointer on the next line when computing `file_size`.

Reply:
> I see it's done further down; let's move it up so it's done before dereferencing blob_file_meta on the next line.

Sure